tests/Doctrine/Tests/ORM/Persisters/BasicEntityPersisterTypeValueSqlTest.php

```diff
@@ -203,7 +203,7 @@ public function testDeleteManyToManyUsesTypeValuesSQL(): void
 
         $connection->expects($this->atLeast(2))
             ->method('delete')
-            ->will($this->onConsecutiveCalls(
+            ->withConsecutive(
```

Comment:
Note, `withConsecutive()` will be deprecated in PHPUnit 10 without a replacement (https://github.com/sebastianbergmann/phpunit/issues/4564). The alternative looks quite clumsy:

```php
$expectedArguments = [
    [
        'customtype_parent_friends',
        ['friend_customtypeparent_id' => 1],
        ['integer'],
    ],
    [
        'customtype_parent_friends',
        ['customtypeparent_id' => 1],
        ['integer'],
    ],
];

$connection->expects($this->atLeast(2))
    ->method('delete')
    ->willReturnCallback(function (string $table, array $where, array $types) use (&$expectedArguments): int {
        $currentExpectedArguments = current($expectedArguments);
        next($expectedArguments);
        if ($currentExpectedArguments !== false) {
            self::assertEquals($currentExpectedArguments, [$table, $where, $types]);
        }

        return 0;
    });
```

Let's use the available API for now and rework it if/when it becomes necessary.